include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -235,6 +235,32 @@ struct reduce
     {
         return reduce_impl(__item_id, __global_idx, __n, __local_mem, __has_known_identity<_BinaryOperation1, _Tp>{});
     }
+
+    template <typename _Result>
+    auto
+    apply_init(_Result __result) const
+    {
+        return __result; //no op
+    }
+};
+
+// Reduce on local memory
+template <typename _ExecutionPolicy, typename _BinaryOperation1, typename _Tp>
+struct reduce_init : public reduce<_ExecutionPolicy, _BinaryOperation1, _Tp>
+{
+    _Tp __init;
+
+    reduce_init(_BinaryOperation1 __bi_op, _Tp __i)
```

Comment:
In many many places in this patch we take function arguments by value. Do you think it's really good idea?

Reply:
It  is not actual any more (the code was changed)